protocol/ssh_scanner.go

```diff
@@ -66,4 +66,53 @@ func (sps *SSHProtocolScanner) ScanProtocol(hiddenService string, osc *config.On
 			conn.Close()
 		}
 	}
+	osc.LogInfo(fmt.Sprintf("Checking %s ssh(2222)\n", hiddenService))
+	conn, err := utils.GetNetworkConnection(hiddenService, 2222, osc.TorProxyAddress, osc.Timeout)
+	if err != nil {
+		osc.LogInfo("Failed to connect to service on port 2222\n")
+		report.SSHDetected = false
+		if conn != nil {
+			conn.Close()
+		}
+	} else {
+		// TODO SSH Checking
+		report.SSHDetected = true
+
+		config := &ssh.ClientConfig{
+			HostKeyCallback: func(hostname string, addr net.Addr, key ssh.PublicKey) error {
+				h := md5.New()
+				h.Write(key.Marshal())
+
+				fBytes := h.Sum(nil)
+				fingerprint := string("")
+				for i := 0; i < len(fBytes); i++ {
+					if i+1 != len(fBytes) {
+						fingerprint = fmt.Sprintf("%s%0.2x:", fingerprint, fBytes[i])
+					} else {
+						fingerprint = fmt.Sprintf("%s%0.2x", fingerprint, fBytes[i])
+					}
+				}
+				report.SSHKey = fingerprint
+				osc.LogInfo(fmt.Sprintf("Found SSH Key %s\n", fingerprint))
+				// We don't want to continue
+				return errors.New("error")
+			},
+		}
+		ssh.NewClientConn(conn, hiddenService+":2222", config)
+		if conn != nil {
+			conn.Close()
+		}
+		conn, err = utils.GetNetworkConnection(hiddenService, 2222, osc.TorProxyAddress, osc.Timeout)
+		if err == nil {
+			reader := bufio.NewReader(conn)
+			banner, err := reader.ReadString('\n')
+			if err == nil {
+				report.SSHBanner = banner
+				osc.LogInfo(fmt.Sprintf("Found SSH Banner: %s", banner))
+			}
+		}
+		if conn != nil {
+			conn.Close()
+		}
+	}
```

Comment:
Same as above, let's not repeat a bunch of code, let's make port number configurable instead.